pkg/sqlcmd/format_ascii.go

```diff
@@ -0,0 +1,174 @@
+package sqlcmd
+
+import (
+	"database/sql"
+	"os"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/microsoft/go-sqlcmd/internal/color"
+	"golang.org/x/term"
+)
+
+type asciiFormatter struct {
+	*sqlCmdFormatterType
+	rows [][]string
+}
+
+func NewSQLCmdAsciiFormatter(vars *Variables, removeTrailingSpaces bool, ccb ControlCharacterBehavior) Formatter {
+	return &asciiFormatter{
+		sqlCmdFormatterType: &sqlCmdFormatterType{
+			removeTrailingSpaces: removeTrailingSpaces,
+			format:               "ascii",
+			colorizer:            color.New(false),
+			ccb:                  ccb,
+			vars:                 vars,
+		},
+	}
+}
+
+func (f *asciiFormatter) BeginResultSet(cols []*sql.ColumnType) {
+	f.sqlCmdFormatterType.BeginResultSet(cols)
+	f.rows = make([][]string, 0)
+}
+
+func (f *asciiFormatter) AddRow(row *sql.Rows) string {
+	values, err := f.scanRow(row)
+	if err != nil {
+		f.mustWriteErr(err.Error())
+		return ""
+	}
+	f.rows = append(f.rows, values)
+	if len(values) > 0 {
+		return values[0]
+	}
+	return ""
+}
+
+func (f *asciiFormatter) EndResultSet() {
+	if len(f.rows) > 0 || len(f.columnDetails) > 0 {
+		f.printAsciiTable()
+	}
+	f.rows = nil
+}
+
+func (f *asciiFormatter) printAsciiTable() {
+	colWidths := make([]int, len(f.columnDetails))
+
+	for i, c := range f.columnDetails {
+		colWidths[i] = utf8.RuneCountInString(c.col.Name())
+	}
+
+	for _, row := range f.rows {
```

Comment:
can you build up the width metrics within `AddRow` instead of waiting for the end of the resultset? There could be thousands of rows and this extra iteration over them can take a long time